### ISSUE TITLE
Question !!!

### ISSUE THREAD

**Issue by anath93** [onlogic/Updating-to-Jetpack-6-for-Aetina-Jetson#2]
Is it possible to build real time kernel which I have done already with 5.15 on Jetson developer kit, and finally apply your ./setup.sh script to the Linux_tegra folder and flash ? or will that be experimental ?

**Comment by anath93** [comment]
Also one more question, how to access GPIO registers using the board in program ?(meaning GPIO assignment) For Libgpiod and JetPy

**Comment by colewangsness** [comment]
Hello.
Answer to Q1: I haven't tested it with the RT kernel so it would be experimental.
Answer to Q2: 

- GPIO Pin addresses for Nano and NX systems PIN 1 to PIN 5 on the GPIO/UART/I2C DB15. Manual Reference page 12. https://www.dropbox.com/scl/fi/o4rogjf4r6nldqs6ow7e8/Aetina_AIE-PO22_32_PN32_42_user-manual.pdf?rlkey=d85bauvq4isimlzmu8hwtrq8r&e=1&dl=0
- For Example GPIO_1(GPIO01: PQ.05 453). Depends on what your application expects, Libgpiod appears to generate based on the module, so it would be best to run the detection CLI tool as listed in the readme.
-cat /sys/kernel/debug/gpio will line up with the above addresses.

**Comment by anath93** [comment]
@colewangsness Is this the file for pinmux changes (tegra234-mb1-bct-pinmux-p3767-dp-a03.dtsi in Linux_For_Tegra/bootloader/generic/BCT/ ?

**Comment by colewangsness** [comment]
@anath93 It appears so, considering that is one of the only files in the patch folder. However, it doesn't look like the system enumeration matches the name (if I am reading this correctly). I'm only seeing pin 4 (GPIO13) and the address doesn't match what the manual and gpiochip0 table show. So I wouldn't necessarily use that for application development.

I would still recommend using gpioinfo command for lines/names if you are still planning to use libgpiod. I've already listed out those in the Jetson GPIO examples section.

**Comment by anath93** [comment]
https://github.com/user-attachments/assets/a8b35845-65fa-4a06-b129-47eaca360eef

Please see this GPIO behavior, its not able to support the current, but based on Aetina it should, 
![image](https://github.com/user-attachments/assets/c0165f72-e959-4b3e-bceb-07edae36f83b)

Commands used to test these, 

![image](https://github.com/user-attachments/assets/2a090bb5-d2c4-4d38-8ce0-69a65c774466)

**Comment by colewangsness** [comment]
@anath93 I tried running the same commands as you've shown above with pins GPIO1 and GND connected. While I did not necessarily see the same behavior initially. When I attached my ground probe to the chassis instead of the GND wire, I did see the same exact issue you have shown above. That could be interfering with the push up/pull down. Our support team is going to connect you with Aetina for further troubleshooting.
